pandas/util/_decorators.py

```diff
@@ -212,6 +212,61 @@ def wrapper(*args, **kwargs):
     return _deprecate_kwarg
 
 
+def deprecate_nonkeyword_arguments(
+    version: str, allowed_args=None, stacklevel: int = 2
+) -> Callable:
+    """
+    Decorator to deprecate a use of non-keyword arguments of a function.
+
+    Parameters
+    ----------
+    version : str
+        The version in which positional arguments will become
+        keyword-only.
+
+    allowed_args : list, int or None, default=None
+        In case of list, it must be the list of names of some
+        first arguments of the decorated functions that are
+        OK to be given as positional arguments. In case of an
+        integer, this is the number of positional arguments
+        that will stay positional. In case of None value,
+        defaults to list of all arguments not having the
+        default value.
+
+    stacklevel : int, default=2
+        The stack level for warnings.warn
+    """
+
+    def decorate(func):
+        if allowed_args is None:
+            spec = inspect.getfullargspec(func)
+            allow_args = spec.args[: -len(spec.defaults)]
+        else:
+            allow_args = allowed_args
+
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            if isinstance(allow_args, int) and len(args) > allow_args:
+                msg = (
+                    "After version %s all arguments of %s "
```

Comment:
version -> pandas version.

Use `{}`-style string templates.

Reply:
Ok, done it. I also added a separate function to form a better looking warning message depending on the number of arguments, but so far I only tested it manually. Is there a proper way to test how exactly does the produced warning look?